ocaml/xapi/xapi_pif.ml

```diff
@@ -443,8 +443,17 @@ let introduce_internal
   (* return ref of newly created pif record *)
   pif
 
+let assert_no_clustering_enabled ~__context ~network =
+  (Db.Cluster.get_refs_where ~__context
+    ~expr:Db_filter_types.(Eq(Field "network", Literal (Ref.string_of network))))
+    |> function
+    | []   -> ()
+    | _::_ -> raise Api_errors.(Server_error (clustering_enabled_on_network, [Ref.string_of network]))
```

Comment:
We still want to allow IP address changes when in maintenance mode though, where clustering is temporarily disabled on a host. I think this function also needs to take a host parameter, and then use `Xapi_clustering.is_clustering_disabled_on_host`. If clustering is disabled allow the ip to be changed, otherwise forbid it. 
You can test this with XenCenter by putting one host into maintenance mode, and then trying to change the IP. XenCenter should internally disable the cluster host, change IP, enable it again.
Or if you want to do it from the cli you can `xe cluster-host-disable`.

Reply:
Should the host check occur before checking for clusters?

Reply:
Yes